gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ConformanceProblemsContainer.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.conformance;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.zowe.apiml.message.api.ApiMessage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+
+/**
+ * Java class that is used to keep track of found conformance issues
+ */
+public class ConformanceProblemsContainer extends HashMap<String, ArrayList<String>> {
```

Comment:
does it have to extend the map? Or could it use the collection as its state? (and have more feature-related method names for example)